src/EventListener/DeserializeListener.php

```diff
@@ -94,12 +98,15 @@ public function onKernelRequest(GetResponseEvent $event)
             $context[AbstractNormalizer::OBJECT_TO_POPULATE] = $data;
         }
 
-        $request->attributes->set(
-            'data',
-            $this->serializer->deserialize(
-                $requestContent, $context['input_class'], $format, $context
-            )
+        $data = $this->serializer->deserialize(
+            $requestContent, $context['input']['class'] ?? $context['resource_class'], $format, $context
         );
+
+        if (null !== $this->dataTransformer && $this->dataTransformer->supportsTransformation($data, $context)) {
```

Comment:
Maybe could we store if an input class is defined, and skip the call to `supportsTransformation` if not (because this call is on the hot path).

Reply:
By the way, shouldn't this call be done in the normalizer instead of here?